File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__posting_fee.cs
// 実行時間 20ms
using System.Linq;
using static System.Console;

internal class Program
{
    static void Main()
    {
        int[] yxh = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] l = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] m = ReadLine().Split().Select( int.Parse ).ToArray();

        if( yxh[2] <= l[0] ) {
            if( System.Math.Max( yxh[0], yxh[1] ) <= l[1] ) {
                WriteLine( m[0] );
            } else if( yxh[0] + yxh[1] <= l[2] ) {
                WriteLine( m[1] );
            } else {
                WriteLine( m[2] );
            }
        } else {
            if( yxh.Max() <= l[3] ) {
                WriteLine( m[3] );
            } else if( yxh.Sum() <= l[4] ) {
                WriteLine( m[4] );
            } else {
                WriteLine( yxh[0] * yxh[1] * yxh[2] * m[5] );
            }
        }
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__posting_fee.py
# 実行時間 20ms
y, x, h = list(map(int, input().split()))
l = list(map(int, input().split()))
m = list(map(int, input().split()))

if h <= l[0]:
    if max(y, x) <= l[1]:
        print(m[0])
    elif y + x + h <= l[2]:
        print(m[1])
    else:
        print(m[2])
else:
    if max(y, x, h) <= l[3]:
        print(m[3])
    elif y + x + h <= l[4]:
        print(m[4])
    else:
        print(y * x * h * m[5])
